src/hooks/useSubaccount.tsx

```diff
@@ -658,74 +671,191 @@ const useSubaccountContext = ({ localDydxWallet }: { localDydxWallet?: LocalWall
     }
   }, [dispatch]);
 
-  // ------ Trigger Orders Methods ------ //
-  const placeTriggerOrders = useCallback(
-    async ({
-      onError,
-      onSuccess,
-    }: {
-      onError?: (onErrorParams: ErrorParams) => void;
-      onSuccess?: () => void;
-    } = {}) => {
-      const callback = (
-        success: boolean,
-        parsingError?: Nullable<ParsingError>,
-        data?: Nullable<HumanReadableTriggerOrdersPayload>
-      ) => {
-        const placeOrderPayloads = data?.placeOrderPayloads.toArray() ?? [];
-        const cancelOrderPayloads = data?.cancelOrderPayloads.toArray() ?? [];
+  const doCancelOrder = useCallback(
+    async (payload: CancelOrderPayload) => {
+      try {
+        if (!compositeClient) {
+          throw new Error('client not initialized');
+        }
 
-        if (success) {
-          // #OrderOnlyConfirmedCancelViaIndexer
-          // even though trigger orders are probably confirmed canceled, we use indexer as source of truth to trigger order status toast
-          onSuccess?.();
-        } else {
-          const errorParams = getValidErrorParamsFromParsingError(parsingError);
-          onError?.(errorParams);
+        const {
+          subaccountNumber: subaccountNumberToUse,
+          clientId,
+          orderFlags,
+          clobPairId,
+          goodTilBlock,
+          goodTilBlockTime,
+        } = payload;
+
+        const subaccountClientToUse = localDydxWallet
+          ? new SubaccountClient(localDydxWallet, subaccountNumberToUse)
+          : undefined;
 
-          placeOrderPayloads.forEach((payload: HumanReadablePlaceOrderPayload) => {
-            dispatch(
-              placeOrderFailed({
-                clientId: payload.clientId,
-                errorParams,
-              })
-            );
-          });
+        if (subaccountClientToUse == null) {
+          throw new Error('wallet not initialized');
+        }
```

Comment:
would it be better to keep `doCancelOrder` and `doPlaceOrder` consistent and check for localDydxWallet above payload destructure?